### PR TITLE
Remove jQuery from repo wiki creation page

### PR DESCRIPTION
**Pull request by yardenshoham**
- Switched to plain JavaScript
- Tested the wiki creation form functionality and it works as before

# Demo using JavaScript without jQuery
action

### CODE REVIEW

File: web_src/js/utils/dom.js [hunk 1]
```diff
@@ -227,3 +227,20 @@ export function initSubmitEventPolyfill() {
...
+export function isVisible(element) {
+  if (!element) return false;
+
+  const style = window.getComputedStyle(element);
+  return style.width !== '0' &&
+           style.height !== '0' &&
+           style.opacity !== '0' &&
+           style.display !== 'none' &&
+           style.visibility !== 'hidden';
+}
```
Comment (silverwind) [inline web_src/js/utils/dom.js:241]: Maybe just make it match jQuery instead?

https://github.com/jquery/jquery/blob/bf11739f6c6926bc9bc1b5a1460505d3b7ef8b01/src/css/hiddenVisibleSelectors.js#L8-L10

Comment (yardenshoham) [reply web_src/js/utils/dom.js:241]: Done https://github.com/go-gitea/gitea/pull/29271/commits/f695a012f2169b7fd2843894e19a35592eea913c

File: web_src/js/utils/dom.js [hunk 2]
```diff
@@ -227,3 +227,15 @@ export function initSubmitEventPolyfill() {
   document.body.addEventListener('click', submitEventPolyfillListener);
   document.body.addEventListener('focus', submitEventPolyfillListener);
 }
+
+/**
+ * Check if an element is visible.
```
Comment (silverwind) [inline web_src/js/utils/dom.js]:
```suggestion
 * Check if an element is visible, equivalent to jQuery's `:visible` pseudo.
```

**Review by silverwind — APPROVED** [review]

**Review by delvh — APPROVED** [review]

File: web_src/js/features/comp/ComboMarkdownEditor.js [hunk 3]
```diff
@@ -14,19 +14,18 @@ let elementIdCounter = 0;
 
 /**
  * validate if the given textarea is non-empty.
- * @param {jQuery} $textarea
+ * @param {HTMLElement} textarea - The textarea element to be validated.
  * @returns {boolean} returns true if validation succeeded.
  */
-export function validateTextareaNonEmpty($textarea) {
+export function validateTextareaNonEmpty(textarea) {
```
Comment (wxiaoguang) [inline web_src/js/features/comp/ComboMarkdownEditor.js:20]: IIRC there is not only one call to this function.

Does these PR changes all of the callers?

File: web_src/js/features/comp/ComboMarkdownEditor.js [hunk 4]
```diff
@@ -14,19 +14,18 @@ let elementIdCounter = 0;
...
   // When using EasyMDE, the original edit area HTML element is hidden, breaking HTML5 input validation.
   // The workaround (https://github.com/sparksuite/simplemde-markdown-editor/issues/324) doesn't work with contenteditable, so we just show an alert.
-  if (!$textarea.val()) {
-    if ($textarea.is(':visible')) {
-      $textarea.prop('required', true);
-      const $form = $textarea.parents('form');
-      $form[0]?.reportValidity();
+  if (!textarea.value) {
+    if (isVisible(textarea)) {
+      textarea.setAttribute('required', 'true');
```
Comment (wxiaoguang) [inline web_src/js/features/comp/ComboMarkdownEditor.js]: `el.required = true`

https://developer.mozilla.org/en-US/docs/Web/API/HTMLTextAreaElement

File: web_src/js/features/comp/ComboMarkdownEditor.js [hunk 5]
```diff
@@ -14,19 +14,18 @@ let elementIdCounter = 0;
...
+      const form = textarea.closest('form');
+      form?.reportValidity();
     } else {
-      // The alert won't hurt users too much, because we are dropping the EasyMDE and the check only occurs in a few places.
```
Comment (wxiaoguang) [inline web_src/js/features/comp/ComboMarkdownEditor.js:29]: This comment should be kept, because the code blow doesn't look that straightforward.

**Review by wxiaoguang — CHANGES_REQUESTED** [review]
Need more checks. Feel free to dismiss this request when the concerns are addressed.

File: web_src/js/utils/dom.js [hunk 6]
```diff
@@ -227,3 +227,15 @@ export function initSubmitEventPolyfill() {
...
   document.body.addEventListener('focus', submitEventPolyfillListener);
 }
+
+/**
+ * Check if an element is visible, equivalent to jQuery's `:visible` pseudo.
+ * Note: This function doesn't account for all possible visibility scenarios.
+ * @param {HTMLElement} element The element to check.
+ * @returns {boolean} True if the element is visible.
+ */
+export function isVisible(element) {
```
Comment (wxiaoguang) [inline web_src/js/utils/dom.js]: Maybe it should be called "isElemVisible" to match "showElem" and "hideElem" (a nit)

File: web_src/js/features/comp/ComboMarkdownEditor.js (hunk 3 above)
Comment (yardenshoham) [reply web_src/js/features/comp/ComboMarkdownEditor.js:20]: I don't see any other callers
image

File: web_src/js/utils/dom.js (hunk 6 above)
Comment (yardenshoham) [reply web_src/js/utils/dom.js]: Done https://github.com/go-gitea/gitea/pull/29271/commits/f658668951a7934f5e362b0b0d9526b67ecf17b2

File: web_src/js/features/comp/ComboMarkdownEditor.js (hunk 5 above)
Comment (yardenshoham) [reply web_src/js/features/comp/ComboMarkdownEditor.js:29]: Done https://github.com/go-gitea/gitea/pull/29271/commits/f658668951a7934f5e362b0b0d9526b67ecf17b2

File: web_src/js/features/comp/ComboMarkdownEditor.js (hunk 4 above)
Comment (yardenshoham) [reply web_src/js/features/comp/ComboMarkdownEditor.js]: Done https://github.com/go-gitea/gitea/pull/29271/commits/f658668951a7934f5e362b0b0d9526b67ecf17b2

File: web_src/js/features/comp/ComboMarkdownEditor.js (hunk 3 above)
Comment (wxiaoguang) [reply web_src/js/features/comp/ComboMarkdownEditor.js:20]: I can see 2 on the main branch.

image

Comment (yardenshoham) [reply web_src/js/features/comp/ComboMarkdownEditor.js:20]: No idea why my editor lied to me... Done https://github.com/go-gitea/gitea/pull/29271/commits/8d74e012acea66e18948f899b85b5fe138b6b494

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**